src/azure-cli-core/azure/cli/core/commands/command_operation.py

```diff
@@ -236,11 +239,17 @@ def handler(self, command_args):  # pylint: disable=too-many-locals, too-many-st
             if no_wait_param:
                 setterargs[no_wait_param] = command_args[no_wait_param]
 
-        if self.setter_arg_name == 'parameters':
-            result = cached_put(self.cmd, setter, **setterargs)
-        else:
-            result = cached_put(self.cmd, setter, setterargs[self.setter_arg_name],
-                                setter_arg_name=self.setter_arg_name, **setterargs)
+        try:
+            if self.setter_arg_name == 'parameters':
+                result = cached_put(self.cmd, setter, **setterargs)
+            else:
+                result = cached_put(self.cmd, setter, setterargs[self.setter_arg_name],
+                                    setter_arg_name=self.setter_arg_name, **setterargs)
+        except SerializationError:
+            if json_parse_fail_args:
+                logger.warning("JSON parse fail for '--add' argument value '%s'. If you want to pass a JSON string, "
+                               "read more in %s", ' '.join(json_parse_fail_args), ge_update_arg_ref)
```

Comment:
You may directly show the exception raised by `shell_safe_json_parse` as warning. No need to make your own warning message.